### PR TITLE
feat: add getDeviceId function and show deviceId on example app

### PR DESCRIPTION
**Pull request by yuhao900914**

### CODE REVIEW

**Review by haoliu-amp — APPROVED** [review]

File: example/src/SdkMethodSections/DeviceIDSection.tsx [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
+  };
+
+  const getDeviceId = React.useCallback(async () => {
+    const deviceId = await amplitude.getDeviceId();
+    setCurrentDeviceId(deviceId);
+    updateLoading(false);
+  }, [amplitude]);
+
+  React.useEffect(() => {
+    getDeviceId();
```
Comment (kelsonpw) [inline example/src/SdkMethodSections/DeviceIDSection.tsx:23]: Should this be checking if loading is true?

Comment (kelsonpw) [reply example/src/SdkMethodSections/DeviceIDSection.tsx:23]: It's a tad confusing adding dependencies to the effect if they aren't used with the effect.  If absolutely required the best thing to do is add an explicit and named comment detailing why we are adding extra values to the dependency array of a hook

File: example/src/SdkMethodSections/DeviceIDSection.tsx [hunk 2]
```diff
@@ -1,20 +1,32 @@
...
 
   return (
     <SdkSectionLayout heading={'Device ID'}>
-      <Button
-        title={'Regenerate Device ID'}
-        onPress={() => {
-          amplitude.regenerateDeviceId();
-        }}
-      />
+      <Input value={currentDeviceId} editable={false} />
```
Comment (kelsonpw) [inline example/src/SdkMethodSections/DeviceIDSection.tsx:28]: Might be good to wrap in a view without pointer events, ie https://stackoverflow.com/a/54127319

File: example/src/SdkMethodSections/DeviceIDSection.tsx [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+import { Button, Input } from 'react-native-elements';
 import { SdkSectionLayout } from './SdkSectionLayout';
+import { useAmplitude } from '../utils/amplitude';
 
 export const DeviceIDSection = () => {
   const amplitude = useAmplitude();
+  const [currentDeviceId, setCurrentDeviceId] = React.useState('');
+  const [loading, updateLoading] = React.useState(false);
+
+  const handleOnPress = () => {
```
Comment (kelsonpw) [inline example/src/SdkMethodSections/DeviceIDSection.tsx:11]: Can we give this a more contextual name like `handleOnPressRegenerateDeviceId` or `handleRegenerateDeviceId`

**Review by kelsonpw — APPROVED** [review]
Please check and make sure no bugs are introduced to the example

**Comment by kelsonpw** [comment]
Nice work!

**Pull request merged**